### DISCUSSION TITLE
How to insert Inline Code inside a Role Content (if it is possible)?

### DISCUSSION THREAD

**Discussion by dbitouze** [executablebooks/meta#886] [Q&A]
According to [MyST documentation](https://myst-parser.readthedocs.io/en/latest/syntax/roles-and-directives.html#roles-an-in-line-extension-point), the Role syntax is:
```markdown
{role-name}`role content`
```
But also according to [MyST documentation](https://myst-parser.readthedocs.io/en/latest/syntax/syntax.html#commonmark), the Inline Code syntax is:
```markdown
`code`
```
But, obviously, 
```markdown
{role-name}`role content with some `code` inside`
```
is wrong.

Hence my question: how to insert Inline Code inside a Role Content (if it is possible)?

**Comment by chrisjsewell** [comment]
```

{role-name}``role content with some `code` inside``

```

**Reply by dbitouze** [reply]
> ```
> 
> {role-name}``role content with some `code` inside``
> ```

Thanks! But, AFAICS, it just displays the back ticks around “code” (the later being ordinary displayed): 

> role content with some \`code\` inside

**Reply by chrisjsewell** [reply]
It depends on what the role is, whether it parses the content as Markdown or not

**Reply by dbitouze** [reply]
> It depends on what the role is, whether it parses the content as Markdown or not

I'm not sure about this. Here is how the role is defined:

```python
# conf.py
extensions = [
    'sphinx.ext.extlinks',
    'myst_parser',
]
extlinks = {
    'faquk': ('https://texfaq.org/%s','%s'), # :faquk:`FAQ-LaTeX3` (for instance)
}
```
and here is how I would like to use it:
```markdown
{faquk}`\normalsize not defined <FAQ-normalszmiss>`
```
with “\normalsize” in inline code: `\normalsize`.

**Reply by choldgraf** [reply]
I don't believe that this particular role parses its content for formatting, which is why this doesn't work. For example if you do the same but with other rST formatting syntax, they show up directly as well and are not formatted. E.g., try:

```
{faquk}` \normalsize *not* defined <FAQ-normalszmiss>`
```

The `*` will still show up and `not` is not italicized.

**Reply by dbitouze** [reply]
> I don't believe that this particular role parses its content for formatting, which is why this doesn't work.

OK so I should probably ask on Sphinx side.

**Comment by dbitouze** [comment]
Related question: How to insert back ticks inside a Role Content (if it is possible)?

**Reply by choldgraf** [reply]
If the role is started / ended with double backticks, then you can insert single backticks inside and they'll be parsed as part of the content.

For example:

```
{faquk}`` \normalsize `not` defined <FAQ-normalszmiss>``
```

you can even do:

```
{faquk}``` \normalsize ``not`` defined <FAQ-normalszmiss>```
```

whereas this does **not** work:

```
{faquk}` \normalsize `not` defined <FAQ-normalszmiss>`
```

**Reply by dbitouze** [reply]
Sigh... I gave the solution to this very question some posts above: sorry for the noise!
